app/Http/Controllers/BookController.php

```diff
@@ -87,4 +87,46 @@ public function claimBook(int $id, Request $request)
             'data'=> $book
         ]);
     }
+
+    public function returnBook(int $id, Request $request)
+    {
+        $request->validate([
+            'email'=>'required | string'
+        ]);
+
+        $book = $this->book->find($id);
+
+        if(!$book){
+            return response()->json([
+                'message'=> "Book {$id} not found",
+                'success'=> false
+            ],404);
+        }
+
+        if ($book->claimed == 0){
+            return response()->json([
+                'message'=> "Book {$id} is not claimed",
+                'success'=>false
+            ], 400);
+        }
+
+        $exists = $book->where('claimed_by_email', '=', $request->email)->exists();
```

Comment:
This check is running an extra query that isnt really needed - since you already have the $book you could just check if $book->claimed_by_email == $request->email